ddtrace/tracer/option.go

```diff
@@ -829,3 +829,42 @@ func StackFrames(n, skip uint) FinishOption {
 		cfg.SkipStackFrames = skip
 	}
 }
+
+// UserTagOption represents a function that can be provided as a parameter to SetUser.
+// This is used to normalize user tags key/value pairs.
+type UserTagOption func() (key string, val interface{})
```

Comment:
No strong opinion on this, but if you want the 100% Go way of doing an option, then it should be a function closure to be executed on the value to modify.

So this type:
```go
func(span Span)
```

and for example:

```go
func WithUserName(name string) UserOption {
	return func(span Span) {
		span.SetTag("usr.name", name)
	}
}
```

After thoughts: it's actually "faster" because it no longer copies the return values on the stack.